sympy/functions/elementary/miscellaneous.py

```diff
@@ -391,14 +391,12 @@ def __new__(cls, *args, **assumptions):
                 args = frozenset(cls._new_args_filter(args))
             except ShortCircuit:
                 return cls.zero
-        else:
-            args = frozenset(args)
-
-        if evaluate:
             # remove redundant args that are easily identified
             args = cls._collapse_arguments(args, **assumptions)
             # find local zeros
-            args = cls._find_localzeros(args, **assumptions)
+            args = frozenset(cls._find_localzeros(args, **assumptions))
+        else:
+            args = frozenset(args)
```

Comment:
```suggestion
            args = cls._find_localzeros(args, **assumptions)
        args = frozenset(args)
```

Reply:
Yeah, became clear for when one saw the diffs.